dev-libs/libmspack/libmspack-0.6~alpha.recipe

```diff
@@ -13,27 +13,43 @@ version of any particular compressor or decompressor, use the \
 mspack_version() function. The UNIX library interface version is defined as \
 the highest-versioned library component."
 HOMEPAGE="https://www.cabextract.org.uk/libmspack/"
-COPYRIGHT="2000-2015 Stuart Caie"
+COPYRIGHT="2000-2017 Stuart Caie"
 LICENSE="GNU LGPL v2"
-REVISION="2"
+REVISION="1"
 SOURCE_URI="$HOMEPAGE/libmspack-${portVersion/\~/}.tar.gz"
-CHECKSUM_SHA256="8967f275525f5067b364cee43b73e44d0433668c39f9376dfff19f653d1c8110"
+CHECKSUM_SHA256="1edbee82accb28e679ab538f803aab7a5a569e4102ccf1715b462b1bd915f921"
 SOURCE_DIR="libmspack-${portVersion/\~/}"
 
+SOURCE_FILENAME_2="libmspack-${portVersion/\~/_}-fix-tests.patch"
+srcGitRev_2="831379877f8c646c5643dd98305afdb5a6b4ca6c"
+SOURCE_URI_2="https://gitweb.gentoo.org/repo/gentoo.git/plain/dev-libs/libmspack/files/$SOURCE_FILENAME_2?id=$srcGitRev_2#noarchive"
+CHECKSUM_SHA256_2="ab1bbe6b469316c320507f1b13aee36fe06b3732949c52b8e5e0f8dd4c3cf8a1"
```

Comment:
You could grab the upstream commit instead: https://github.com/kyz/libmspack/commit/deb65056f436a4ce06946c4dde270f533e3cab4d

Reply:
Since the fix is only for the tests, I'd rather stick to the stable release.
Dropping these 4 lines and the `PATCH()` function (once the next release is out) will be better than switching to a snapshot and, later, switching back to a stable release.
IMHO the only benefit in swithing to a snapshot would be to get rid of the `PATCH()` function and the `*_2` variables, but we know that these are only temporary.

Reply:
I mean downloading the commit (add `.patch` to the end of commit uri) then do `git am` to the source then `haikuporter -e`

Pros:
- No need to fetch an external patch

Cons:
- Bigger diff

An alternative is to fetch the patch directly from upstream.

Reply:
Hmm, the "grab the patch from Gentoo" method is more generic than the `git am` + `haikuporter -e` method, and imho it has the big advantage of adding a constant number of lines, no matter how big the patch is.
The other advantage is that it makes it clear that the patch is not from us. This also means we credit Gentoo for their nice work :)